OpenRA.Mods.Common/Traits/PaletteEffects/GlobalLightingPaletteEffect.cs

```diff
@@ -0,0 +1,82 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Used for day/night effects.")]
+	class GlobalLightingPaletteEffectInfo : ITraitInfo
+	{
+		[Desc("Do not modify graphics that use any palette in this list.")]
+		public readonly string[] ExcludePalettes = { "cursor", "chrome", "colorpicker", "fog", "shroud", "alpha" };
+
+		[Desc("Only modify graphics that use any of these palettes. If this list has members `ExcludePalettes` is ignored.")]
+		public readonly string[] OnlyPalettes = { };
+
+		[Desc("Do not modify graphics that start with these letters.")]
+		public readonly string[] ExcludePalettePrefixes = { };
+
+		[Desc("Only modify graphics that start with these letters.")]
+		public readonly string[] OnlyPalettePrefixes = { };
```

Comment:
I more meant the combination of a specific-name blacklist and two separate whitelists, which is rather ugly.

It would be more consistent with the other traits if this provided two blacklists (actually most consistent with two *white*lists, but that is straying well outside the scope of this feature since it affects all of the modifier traits): `ExcludePalettes` and `ExcludePlayerPalettes`.